lhctl/cmd/user_task_run.go

```diff
@@ -151,6 +151,83 @@ var getUserTaskRunCmd = &cobra.Command{
 	},
 }
 
+var saveUserTaskRunProgressCmd = &cobra.Command{
+	Use:   "saveUserTaskRun",
+	Short: "Save progress on UserTaskRuns",
+	Long: `Given a provied WfRunId and UserTaskGuid, this utility allows you
+to save current progress on a UserTask before executing the it.
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		wfRunId, _ := cmd.Flags().GetString("wfRunId")
+		userTaskGuid, _ := cmd.Flags().GetString("userTaskGuid")
+
+		saveUserTaskRunProgress := &model.SaveUserTaskRunProgressRequest{
+			Results: make(map[string]*model.VariableValue),
+			UserTaskRunId: &model.UserTaskRunId{
+				WfRunId:      common.StrToWfRunId(wfRunId),
+				UserTaskGuid: userTaskGuid,
+			},
+		}
+
+		// First, get the UserTaskRun.
+		client := getGlobalClient(cmd)
+		userTaskRun, _ := getUserTaskRun(cmd, wfRunId, userTaskGuid, &client)
+
+		if userTaskRun.Notes != nil {
+			fmt.Println("\nNotes: " + *userTaskRun.Notes + "\n")
+		}
+
+		// Next, get the UserTaskDef.
+		userTaskDef, err := getUserTaskDef(cmd, userTaskRun, &client)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// Next, prompt for the userId.
+		userIdVarVal, err := promptFor(
+			"Enter the userId of the person completing the task",
+			model.VariableType_STR,
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+		saveUserTaskRunProgress.UserId = userIdVarVal.GetStr()
+
+		for _, field := range userTaskDef.Fields {
+			fmt.Println("\nField: ", field.DisplayName)
+			if field.Description != nil {
+				fmt.Println(*field.Description)
+			}
+			resultVal, err := promptFor("Please enter the response for this field ("+field.Type.String()+")", field.Type)
+			if err != nil {
+				log.Fatal(err)
+			}
+			saveUserTaskRunProgress.Results[field.Name] = resultVal
+		}
+
+		fmt.Println("Select an assignment policy value:")
+		for k, v := range model.SaveUserTaskRunProgressRequest_SaveUserTaskRunAssignmentPolicy_name {
+			fmt.Printf("%d: %s\n", k, v)
+		}
+
+		assignmentPolicy, err := promptFor(
+			"Enter the number corresponding to your choice: ",
+			model.VariableType_INT,
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+		saveUserTaskRunProgress.Policy = model.SaveUserTaskRunProgressRequest_SaveUserTaskRunAssignmentPolicy(assignmentPolicy.GetInt())
```

Comment:
Most of this code comes from `executeUserTaskRunCmd`, so it is now duplicated.

If we want to spend more time on this, I can refactor and move duplicate code to functions. I don't think this is worth our time right now but let me know your thoughts.

Reply:
That's not a huge issue, but @eduwercamacaro may think differently. The important thing is the verbiage should be `lhctl save userTaskRun` rather than `lhctl saveUserTaskRun`